### PR TITLE
[script.module.libMediathek2] v1.2.0

### PR DESCRIPTION
**Pull request by prof-membrane**
### Description
This is a small framework, used in my "Mediathek" style add-ons. It serves as a layer between Kodi and the content scraper. This allows me to develop the add-ons faster and lets me use the scrapers for other purposes.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a sinlge commit with using the following style: [sciprt.foo.bar] v1.0.0

### CODE REVIEW

File: script.module.libMediathek2/lib/libMediathek2Listing.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import xbmcaddon
+import xbmcvfs
+import re
+from datetime import datetime,timedelta
+import time
+
+from libMediathek2Utils import clearString
+
+icon = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')+'/icon.png').decode('utf-8')
+fanart = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')+'/fanart.jpg').decode('utf-8')
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Listing.py]: getAddonInfo("Icon") and getAddonInfo("fanart") might make more sense here ;)

File: script.module.libMediathek2/lib/libMediathek2Listing.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+def addEntries(l):
+	lists = []
+	for dict in l:
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Listing.py]: dict is quite a dangerous name to choose since python uses that name for dictionaries.

**Comment by prof-membrane** [comment]
Thanks for the input. Is there something else?

File: script.module.libMediathek2/lib/libMediathek2Listing.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			dict['aired'],dict['airedtime'] = _airedISO8601(dict)
+			
+		if 'type' in dict and dict['type'] == 'date' and 'airedtime' in dict:
+			dict["name"] = '(' + str(dict["airedtime"]) + ') ' + dict["name"]
+		elif 'type' in dict and dict['type'] == 'date' and 'time' in dict:
+			dict["name"] = '(' + str(dict["date"]) + ') ' + dict["name"]
+			
+		ilabels = {
+			"Title": clearString(dict.get('name','')),
+			"Plot": clearString(dict.get('plot','')),
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Listing.py]: please also always set the mediatpe for movies/episodes etc.

File: script.module.libMediathek2/lib/libMediathek2PremadeDirs.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		day = date.today() - timedelta(i)
+		dict['name'] = weekdayDict[day.strftime("%w")]
+		dict['datum']  = str(i)
+		dict['mode'] = mode
+		dict['type'] = 'dir'
+		l.append(dict)
+		i += 1
+	return l
+
+def setView(viewMode):
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2PremadeDirs.py]: this setView stuff is hacky, I´d prefer to have it removed.

File: script.module.libMediathek2/lib/libMediathek2Ttml2Srt.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+					else:
+						part = part.replace('<'+entry+'>','')
+				
+
+				buffer += str(i) + '\n'
+				buffer += begin+" --> "+end+"\n"
+				buffer += part + '\n\n'
+				i+=1
+		
+		f = xbmcvfs.File(subFile, 'w')
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Ttml2Srt.py]: use with statement for opening files.

File: script.module.libMediathek2/lib/libMediathek2Ttml2Srt.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+		os.remove(subFile)
+	try:
+		content = utils.getUrl(url)
+	except:
+		content = ""
+	if content:
+		dict = _stylesSetup(re.compile('<styling>(.+?)</styling>', re.DOTALL).findall(content)[0])
+		matchLine = re.compile('<p id=".+?" begin="1(.+?)" end="1(.+?)".+?style="(.+?)">(.+?)</p>', re.DOTALL).findall(content)
+		#fh = open(subFile, 'a')
+		f = xbmcvfs.File(subFile, 'w')
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Ttml2Srt.py]: same here.

File: script.module.libMediathek2/lib/libMediathek2Utils.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+def clearString(s):
+	s = s.replace("&lt;", "<").replace("&gt;", ">").replace("&amp;", "&").replace("&#034;", "\"").replace("&#039;", "'").replace("&quot;", "\"").replace("&szlig;", "ß").replace("&ndash;", "-")
+	s = s.replace("&Auml;", "Ä").replace("&Uuml;", "Ü").replace("&Ouml;", "Ö").replace("&auml;", "ä").replace("&uuml;", "ü").replace("&ouml;", "ö").replace("&eacute;", "é").replace("&egrave;", "è")
+	s = s.replace("&#x00c4;","Ä").replace("&#x00e4;","ä").replace("&#x00d6;","Ö").replace("&#x00f6;","ö").replace("&#x00dc;","Ü").replace("&#x00fc;","ü").replace("&#x00df;","ß")
+	s = s.replace("&apos;","'").strip()
+	return s
+		
+	
+def f_open(path):
+	f = xbmcvfs.File(path)
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Utils.py]: and here.

File: script.module.libMediathek2/lib/libMediathek2Utils.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+	result = f.read()
+	f.close()
+	return result
+
+def f_write(path,data):
+	print 'writing to '+path
+	f = xbmcvfs.File(path, 'w')
```
Comment (phil65) [inline script.module.libMediathek2/lib/libMediathek2Utils.py]: and here.

File: script.module.libMediathek2/resources/language/English/strings.xml [hunk 9]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<strings>
```
Comment (phil65) [inline script.module.libMediathek2/resources/language/English/strings.xml]: would be cool if you would convert the xml language files to .po format.
xml format is deprecated.

**Comment by phil65** [comment]
That´s all. Thx!

**Comment by phil65** [comment]
oops. :)

**Comment by phil65** [comment]
Please also change your addon id to lowercase-only.

**Comment by MartijnKaijser** [comment]
A module should never have settings. The plugin should request have the settings needed and the module should just do as asked by the plugin and return the information.

File: script.module.libMediathek2/lib/libMediathek2PremadeDirs.py (hunk 4 above)
Comment (MartijnKaijser) [inline script.module.libMediathek2/lib/libMediathek2PremadeDirs.py]: I'd say completely disallow and deprecate it to be set for existing plugins. If they update make sure to mention it should not be included anymore

**Comment by prof-membrane** [comment]
Sorry, this was just a leftover. I'll remove them.

**Pull request closed without merging**